[core-http] Update changelog for March 2020 release (#7562)
Azure/azure-sdk-for-js: sdk/core/core-http/CHANGELOG.md

# Release History

## 1.0.4 (2020-02-28)

- When an operation times out based on the `timeout` configured in the `OperationRequestOptions`, it gets terminated with an error. In this update, the error that is thrown in browser for such cases is updated to match what is thrown in node i.e an `AbortError` is thrown instead of the previous `RestError`. [PR #7159](https://github.com/Azure/azure-sdk-for-js/pull/7159)

> - User name and password in proxy url string are now supported [PR #7211](https://github.com/Azure/azure-sdk-for-js/pull/7211)

Reviewer: ```suggestion
- Providing username and password in the proxy url is now supported [PR #7211](https://github.com/Azure/azure-sdk-for-js/pull/7211)
```

> - Consumers of this library no longer need to reference lib "dom" or add shims for types from `lib.dom.d.ts` when they don't use those types directly. [PR #7500](https://github.com/Azure/azure-sdk-for-js/pull/7500)

Reviewer: ```suggestion
- Removed dependency on `lib.dom.d.ts` so TypeScript users no longer need `lib: ["dom"]` in their tsconfig.json file to use this library.  [PR #7500](https://github.com/Azure/azure-sdk-for-js/pull/7500)
```

Author: Thanks for the suggestions!
